### PR TITLE
halonet pr

### PR DESCRIPTION
**Pull request by xjtu-fgh**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/halonet/halonet_50t_ascend.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+drop_remainder: True
+val_split: val
+# Augmentation config
+image_resize: 256
+scale: [0.08, 1.0]
+ratio: [0.75, 1.333]
+hflip: 0.5
+interpolation: 'bilinear'
+crop_pct: 0.95
+#color_jitter:直接在config改了
```
Comment (SamitHuang) [inline configs/halonet/halonet_50t_ascend.yaml:21]: remove cn comment

File: mindcv/models/halonet.py
Comment (SamitHuang) [inline mindcv/models/halonet.py:1]: Use precommit to refine the code format

File: mindcv/models/halonet.py [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+                                block_size=block_size,
+                                halo_size=halo_size,
+                                depth=depth_list[3],
+                                channel=chs_list[3],
+                                out_channel=chs_list[4],
+                                stride=stride_list[3],
+                                num_head=num_heads[3],
+                                hidden_chs=hidden_chs,
+                                act=act,
+                                downsample=True)
```
Comment (SamitHuang) [inline mindcv/models/halonet.py:576]: Consider putting them into a list by for loop for extendability

File: mindcv/models/halonet.py [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+        v = kv[...,self.dim_head_qk:(self.dim_head_qk + self.dim_head_v)]
+        k = ops.transpose(k,(0,1,3,2)) # [B * self.num_heads, num_blocks, self.dim_head_qk, -1]
+        if self.scale_pos_embed:
+            attn = (ops.matmul(q, k) + self.pos_embed(q)) * self.scale
+        else:
+            pos_embed_q = self.pos_embed(q)
+            part_1 = (ops.matmul(q, k)) * self.scale
+            attn = part_1 + pos_embed_q
+        # attn: B * num_heads, num_blocks, block_size ** 2, win_size ** 2
+        softmax_fn = ops.Softmax(-1)
```
Comment (SamitHuang) [inline mindcv/models/halonet.py:298]: put it in init

**Review by SamitHuang — COMMENTED** [review]
Need revision

**Comment by geniuspatrick** [comment]
@Songyuanwei please help to improve this PR.

File: configs/halonet/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+# ConvNeXt
+> [Scaling Local Self-Attention for Parameter Effificient Visual Backbones](https://arxiv.org/abs/2103.12731)
+
+## Introduction
+
+​		Researchers from Google Research and UC Berkeley have developed a new model of self-attention that can outperform standard baseline models and even high-performance convolutional models.[[1](https://github.com/mindspore-lab/mindcv/diffs/0?base_sha=308825d43f92ea20ccae08b7f388948349664596&branch=6ed77368429ac1b6e6586d8cfcd27c476e6e214f&commentable=true&head_user=xjtu-fgh&name=6ed77368429ac1b6e6586d8cfcd27c476e6e214f&pull_number=713&qualified_name=6ed77368429ac1b6e6586d8cfcd27c476e6e214f&sha1=308825d43f92ea20ccae08b7f388948349664596&sha2=5dcde6e357c4ea1f885474d3f2775d34f25d70b8&short_path=d3e159c&unchanged=expanded&w=false#references)]
```
Comment (Songyuanwei) [inline configs/halonet/README.md:6]: 参考文献引用请参考其他模型readme写法

File: configs/halonet/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Results
+
+Our reproduced model performance on ImageNet-1K is reported as follows.
+
+<div align="center">
+
+| Model      | Context   | Top-1 (%) | Top-5 (%) | Params (M) | Recipe                                                       | Download                                                     |
+| ---------- | --------- | --------- | --------- | ---------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| halonet50t | D910x64-G | 79.528    | 94.788    | 28.59      | [yaml](https://github.com/mindspore-lab/mindcv/blob/main/configs/halonet/halonet_50t_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/halonet/halonet50t-d441ba2c.ckpt) |
```
Comment (Songyuanwei) [inline configs/halonet/README.md:19]: 模型参数是28.59吗？看一下训练日志里面打印的模型参数信息

**Pull request closed without merging**